### PR TITLE
Refactor Edit to Update and modify EditCommand syntax

### LINKED ISSUE #88: Refactor AddressBook's edit functionality syntax to THRIFT's update
- Refactor `edit` instances and associated classes to `update` in THRIFT context.
- Change syntax of `edit` command to THRIFT's `update`.

Increment for #57

### PR DESCRIPTION
**Pull request by lye-jw**
1. Refactors all classes and instances of `edit` to `update`.
2. Modifies syntax of `edit` command to `update` command as listed in UserGuide.

Resolves #88

### CODE REVIEW

**Comment by lye-jw** [comment]
Json Storage is also updated correctly.

**Comment by lye-jw** [comment]
Showing original transaction in status message failed CI due to outdated tests, working on it. In the meantime this PR can be merged if there are no issues.

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -789,9 +800,11 @@ These instructions only provide a starting point for testers to work on; testers
 . Updating a transaction's details and reflecting changes in the list.
 
 .. Prerequisites: List all transactions using the `list` command. The first transaction is `1. [-] Laksa ($10.50) [Lunch][Food]`.
-.. Test case: `update i/1 | n/Fried Rice` +
+.. Test case: `update i/1 n/Fried Rice` +
+   Expected: Details of the updated transaction and original value of updated content shown in status message. Corresponding listed transaction reflects changes. Timestamp in the status bar is updated.
```
Comment (limerencee) [inline docs/DeveloperGuide.adoc]: I think we don't display the timestamp in the status bar, perhaps the last sentence could be removed? Same goes for the other points in this section!

File: docs/UserGuide.adoc [hunk 2]
```diff
@@ -101,28 +101,31 @@ Expected output:
...
 
 Expected output: +
 
 NOTE: Assume 2nd command occurs after 1st
 
-*   `Entry name updated: 1. [-] Curry Laksa ($10.50) [Lunch][Food] +
-Original name was: Laksa`
-*   `Entry cost updated: 1. [-] Curry Laksa ($11) [Lunch][Food] +
-Original cost was: $10.50`
+*   `Updated Transaction: [-] Curry Laksa ($10.50) Date: 09/10/2019 Tags: [Food][Lunch]` +
```
Comment (limerencee) [inline docs/UserGuide.adoc:123]: We could omit showing the original transaction details. If there are mistakes the user could either `undo` or just run `update` again?

File: src/test/java/thrift/logic/commands/AddExpenseCommandTest.java [hunk 3]
```diff
@@ -147,7 +147,7 @@ public void deleteTransaction(Transaction transaction) {
         }
 
         @Override
-        public void setTransaction(Transaction target, Transaction editedTransaction) {
+        public void setTransaction(Transaction target, Transaction updatedTransaction) {
```
Comment (limerencee) [inline src/test/java/thrift/logic/commands/AddExpenseCommandTest.java:150]: Please update this same method under `AddIncomeCommandTest.java` as well, thank you!

**Review by limerencee — CHANGES_REQUESTED** [review]
Looks good! Thanks for your contributions :)
Should be able to merge after the changes are made!

**Review by limerencee — APPROVED** [review]

**Pull request merged**